service-base/src/test/java/org/eclipse/hono/service/VertxBasedHealthCheckServerTest.java

```diff
@@ -158,7 +160,7 @@ void testHealthCheckServerWithSecureAndInsecureEndpoint(final Vertx vertx, final
             .compose(result -> Future.succeededFuture(getWebClient(vertx, server, false)))
             .compose(httpClient -> checkHealth(ctx, httpClient, "/liveness"))
             .compose(httpClient -> checkHealth(ctx, httpClient, "/readiness"))
-            .onComplete(ctx.completing());
+            .onComplete(ctx.succeeding(v -> testsDone.flag()));
```

Comment:
why should this change be necessary?

Reply:
Because of the `registerHealthChecks` invocation before - see obvious.

Reply:
I see